keyboards/ergo42/keymaps/default-underglow/keymap.c

```diff
@@ -0,0 +1,133 @@
+   * | META | LCtrl|   `  |   \  | LAlt | LGUI |Space |   |Space |   '  |   -  |   =  | LEFT | DOWN | RIGHT|
+   * `------------------------------------------------'   `------------------------------------------------'
+   */
+  [SYMB] = LAYOUT( \
+    _______, S(KC_1), S(KC_2), S(KC_3), S(KC_4), S(KC_5), KC_LBRC,    KC_RBRC,    S(KC_6), S(KC_7), S(KC_8), S(KC_9), S(KC_0), _______, \
+    _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, S(KC_9),    S(KC_0),    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, _______, \
+    _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, S(KC_LBRC), S(KC_RBRC), XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, _______, _______, \
+    _______, _______, _______, _______, _______, _______, _______,    _______,    _______, _______, _______, _______, _______, _______ \
+  ),
+
+  /* RGB
+   * ,------------------------------------------------.   ,------------------------------------------------.
+   * |      |      |      |      |      |      |      |   |      |      |      |      |      |      |      |
+   * |------+------+------+------+------+------+------|   |-------------+------+------+------+------+------|
+   * |      |      |      |      |      |      |      |   |      | TOG  | HUI  | SAI  | VAI  |HUANIM|      |
+   * |------+------+------+------+------+------+------|   |------|------+------+------+------+------+------|
+   * |      |      |      |      |      |      |      |   |      | MOD  | HUD  | SAD  | VAD  |RESET |      |
+   * |------+------+------+------+------+------+------|   |------+------+------+------+------+------+------|
+   * |      |      |      |      |      |      |      |   |      |      |      |      |      |=>RGB |      |
+   * `------------------------------------------------'   `------------------------------------------------'
+   */
+  [RGB] = LAYOUT( \
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, \
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, TAP_ANIM,XXXXXXX, \
+    _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RGB_MOD, RGB_HUI, RGB_SAD, RGB_VAD, RGB_RST, XXXXXXX, \
+    XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX \
+  )
+
+};
+
+bool isTapAnim = false;
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case RGB_MOD:
+      if (record->event.pressed) {
+        isTapAnim = false;
+      }
+      break;
+    case RGB_RST:
+      #ifdef RGBLIGHT_ENABLE
+        if (record->event.pressed) {
+          eeconfig_update_rgblight_default();
+          rgblight_enable();
+        }
+      #endif
+      break;
+    case TAP_ANIM:
+      if (record->event.pressed) {
+        isTapAnim = !isTapAnim;
+      }
+      break;
+  }
+
+  if (isTapAnim) {
```

Comment:
For this stuff, if you wanted a "funner" effect, I have a "twinkling" type effect here:

https://github.com/qmk/qmk_firmware/blob/master/users/drashna/rgb_stuff.c#L108-L253

What this does, is each key you hit causes a single LED to "light up" and twinkle basically.   It's a bit complex, but if you wanted to use that instead, then feel free!